src/can.rs

```diff
@@ -20,10 +20,72 @@ use crate::stm32;
 use nb::{self, Error};
 
 use core::sync::atomic::{AtomicU8, Ordering};
+use stm32::can::btr::LBKM_A;
 
 const EXID_MASK: u32 = 0b1_1111_1111_1100_0000_0000_0000_0000;
 const MAX_EXTENDED_ID: u32 = 0x1FFF_FFFF;
 
+/// Options the CAN bus. This is primarily used to set bus timings, but also controls options like enabling loopback or silent mode for debugging.
+/// See  http://www.bittiming.can-wiki.info/#bxCAN for generating the timing parameters for different baud rates and clocks.
+///
+/// Use `CanOpts::default()` to get 250kbps at 32mhz system clock
+pub struct CanOpts {
+    brp: u16,
+    sjw: u8,
+    ts1: u8,
+    ts2: u8,
+    lbkm: LBKM_A,
+}
+
+impl CanOpts {
+    /// Create new `CanOpts` using the default settings from `CanOpts::default()` to get 250kbps at 32mhz system clock.
+    pub fn new() -> CanOpts {
+        CanOpts::default()
+    }
+
+    /// Set the Baud Rate Prescaler. See  http://www.bittiming.can-wiki.info/#bxCAN for generating the timing parameters for different baud rates and clocks.
+    pub fn brp(mut self, brp: u16) -> Self {
+        self.brp = brp;
+        self
+    }
+
+    /// Set the Resynchronisation Jump Width. See  http://www.bittiming.can-wiki.info/#bxCAN for generating the timing parameters for different baud rates and clocks.
+    pub fn sjw(mut self, sjw: u8) -> Self {
+        self.sjw = sjw;
+        self
+    }
+
+    /// Set Time Segment One. See  http://www.bittiming.can-wiki.info/#bxCAN for generating the timing parameters for different baud rates and clocks.
+    pub fn ts1(mut self, ts1: u8) -> Self {
+        self.ts1 = ts1;
+        self
+    }
+
+    /// Set Time Segment Two. See  http://www.bittiming.can-wiki.info/#bxCAN for generating the timing parameters for different baud rates and clocks.
+    pub fn ts2(mut self, ts2: u8) -> Self {
+        self.ts2 = ts2;
+        self
+    }
+
+    /// Enable or disable loopback mode on the CAN device. This is useful for debugging.
+    pub fn lbkm(mut self, lbkm: LBKM_A) -> Self {
```

Comment:
Maybe reexport the `LBKM_A` type, so that the user does not have to dig through the stm32::can abstractions to find the correct enum.